telegram/ext/_applicationbuilder.py

```diff
@@ -0,0 +1,766 @@
+        '_update_queue',
+        '_job_queue',
+        '_persistence',
+        '_context_types',
+        '_application_class',
+        '_application_kwargs',
+        '_concurrent_updates',
+        '_updater',
+    )
+
+    def __init__(self: 'InitApplicationBuilder'):
+        self._token: DVInput[str] = DefaultValue('')
+        self._base_url: DVInput[str] = DefaultValue('https://api.telegram.org/bot')
+        self._base_file_url: DVInput[str] = DefaultValue('https://api.telegram.org/file/bot')
+        self._connection_pool_size: DVInput[int] = DEFAULT_NONE
+        self._proxy_url: DVInput[str] = DEFAULT_NONE
+        self._connect_timeout: ODVInput[float] = DEFAULT_NONE
+        self._read_timeout: ODVInput[float] = DEFAULT_NONE
+        self._write_timeout: ODVInput[float] = DEFAULT_NONE
+        self._pool_timeout: ODVInput[float] = DEFAULT_NONE
+        self._request: DVInput['BaseRequest'] = DEFAULT_NONE
+        self._get_updates_connection_pool_size: DVInput[int] = DEFAULT_NONE
+        self._get_updates_proxy_url: DVInput[str] = DEFAULT_NONE
+        self._get_updates_connect_timeout: ODVInput[float] = DEFAULT_NONE
+        self._get_updates_read_timeout: ODVInput[float] = DEFAULT_NONE
+        self._get_updates_write_timeout: ODVInput[float] = DEFAULT_NONE
+        self._get_updates_pool_timeout: ODVInput[float] = DEFAULT_NONE
+        self._get_updates_request: DVInput['BaseRequest'] = DEFAULT_NONE
+        self._private_key: ODVInput[bytes] = DEFAULT_NONE
+        self._private_key_password: ODVInput[bytes] = DEFAULT_NONE
+        self._defaults: ODVInput['Defaults'] = DEFAULT_NONE
+        self._arbitrary_callback_data: DVInput[Union[bool, int]] = DEFAULT_FALSE
+        self._bot: DVInput[Bot] = DEFAULT_NONE
+        self._update_queue: DVInput[Queue] = DefaultValue(Queue())
+        self._job_queue: ODVInput['JobQueue'] = DefaultValue(JobQueue())
+        self._persistence: ODVInput['BasePersistence'] = DEFAULT_NONE
+        self._context_types: DVInput[ContextTypes] = DefaultValue(ContextTypes())
+        self._application_class: DVInput[Type[Application]] = DefaultValue(Application)
+        self._application_kwargs: Dict[str, object] = {}
+        self._concurrent_updates: DVInput[Union[int, bool]] = DEFAULT_FALSE
+        self._updater: ODVInput[Updater] = DEFAULT_NONE
+
+    def _build_request(self, get_updates: bool) -> BaseRequest:
+        prefix = '_get_updates_' if get_updates else '_'
+        if not isinstance(getattr(self, f'{prefix}request'), DefaultValue):
+            return getattr(self, f'{prefix}request')
+
+        proxy_url = DefaultValue.get_value(getattr(self, f'{prefix}proxy_url'))
+        if get_updates:
+            connection_pool_size = (
+                DefaultValue.get_value(getattr(self, f'{prefix}connection_pool_size')) or 1
+            )
+        else:
+            connection_pool_size = (
+                DefaultValue.get_value(getattr(self, f'{prefix}connection_pool_size')) or 128
```

Comment:
why do we set connection_pool_size to 128? Shouldn't it be the user set value or the default -- 1?

Reply:
Also why have two `HTTPXRequest` instances (One for `get_updates`, other for the rest of the bot methods). Why not just use the same for both

Reply:
`DefaultValue.get_value(getattr(self, f'{prefix}connection_pool_size'))` does get the custom value set by the user, if there is any such that `user_value or 128` will result in `user_value`.
For non-get_updates request we use a large connection pool since we can now handle that via asyncio - in contrast to the current policy of "number of threads". One can discuss where that default should be set  - here on in the signature of `HTTPXRequest`. IMO setting it here makes the separation between the pure networking backend and "where & how it's used" a tiny bit more clear.

Regarding two request instances: At runtime, we'll want to make sure that we never try to make more requests in parallel than we have connections in the pool. This is a bit tricky to do with only one request object without mixing the networking level and the API level - i.e. the `BaseRequest` object would have to know if we're doing a `getUpdates` request or some other request. That's why Noam & I came up with the concept of just using two instances. Note that we only need 1 connection in the pool for `get_updates` in an `Application`, as `Updater` will never make more than 1 request to fetch updates at a time.